yb-voyager/src/mat/plugins/common.go

```diff
@@ -0,0 +1,104 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mat
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pelletier/go-toml"
+	log "github.com/sirupsen/logrus"
+)
+
+type Record map[string]any
+type QueryResult []Record
+
+func LoadQueryResults(pluginName string, exportDir string) (map[string]QueryResult, error) {
+	queryResults := make(map[string]QueryResult)
+	assessmentDataDir := filepath.Join(exportDir, "assessment", "data")
+	files, err := filepath.Glob(filepath.Join(assessmentDataDir, fmt.Sprintf("%s__*.csv", pluginName)))
+	if err != nil {
+		log.Errorf("error listing files in %s for plugin %s: %v", assessmentDataDir, pluginName, err)
+		return nil, fmt.Errorf("error listing files in %s for plugin %s: %w", assessmentDataDir, pluginName, err)
+	}
+
+	for _, file := range files {
+		baseFileName := filepath.Base(file)
+		queryName := strings.TrimSuffix(strings.TrimPrefix(baseFileName, fmt.Sprintf("%s__", pluginName)), ".csv")
+		// fmt.Printf("baseFileName: %s, queryName: %s\n", baseFileName, queryName)
+		// TODO: read file line by line using a reader or read full file once
+		// Read the file
+		bytes, err := os.ReadFile(file)
+		if err != nil {
+			log.Errorf("error opening file %s: %v", file, err)
+			return nil, fmt.Errorf("error opening file %s: %w", file, err)
+		}
+
+		if len(bytes) == 0 {
+			log.Warnf("file %s is empty", file)
+			continue
+		}
+
+		rows := strings.Split(string(bytes), "\n")
+		columns := strings.Split(rows[0], ",")
+		queryResult := make(QueryResult, 0)
+		for i := 1; i < len(rows); i++ {
+			if len(rows[i]) == 0 {
+				continue
+			}
+			row := strings.Split(rows[i], ",")
+			// fmt.Printf("row: %v\n", row)
+			record := make(Record)
+			for j, column := range columns {
+				record[column] = row[j]
+			}
+			queryResult = append(queryResult, record)
+		}
+		queryResults[queryName] = queryResult
+	}
+
+	return queryResults, nil
+}
+
+func LoadUserInput(pluginName string, userInputFpath string) (map[string]any, error) {
+	tomlData, err := os.ReadFile(userInputFpath)
+	if err != nil {
+		log.Errorf("error reading plugin params file %s: %v", userInputFpath, err)
+		return nil, fmt.Errorf("error reading plugin params file %s: %w", userInputFpath, err)
+	}
+
+	tree, err := toml.LoadBytes(tomlData)
+	if err != nil {
+		log.Errorf("error parsing plugin params file %s: %v", userInputFpath, err)
+		return nil, fmt.Errorf("error parsing plugin params file %s: %w", userInputFpath, err)
+	}
+
+	tableNames := []string{"common", pluginName}
+	userInputParams := make(map[string]any)
+	for _, tableName := range tableNames {
+		if !tree.Has(tableName) {
+			log.Warnf("table '%s' not found in plugin params file %s", tableName, userInputFpath)
+			continue
+		}
+		table := tree.Get(tableName).(*toml.Tree)
+		userInputParams[tableName] = table.ToMap()
```

Comment:
Merge the `common` and `<PLUGIN>` sections in one flat k-v map:
`userInputParams = lo.Assign(userInputParams, table.ToMap())`